### PR TITLE
Fixed bugs when switching back to graph view:

### PR DESCRIPTION
**Pull request by vyakunin**
- empty graph was shown
- crashed when tried to add log

Signed-off-by: Inessa <inessa@Inessas-MacBook-Air.local>

### CODE REVIEW

File: android/app/src/main/java/io/peggyjo/peggymeter/HistoryTextFragment.java [hunk 1]
```diff
@@ -43,7 +43,8 @@ private void renderLog(View view, List<LogEntry> logs) {
                 getResources().getString(R.string.smile2),
                 getResources().getString(R.string.smile3),
                 getResources().getString(R.string.smile4)};
-        table.setAdapter(new TableRowAdapter(getContext(), R.layout.listview_item_row, reverse(logs), smileys));
+        table.setAdapter(new TableRowAdapter(getContext(), R.layout.listview_item_row,
+                reverse(logs), smileys));
```
Comment (rustema) [inline android/app/src/main/java/io/peggyjo/peggymeter/HistoryTextFragment.java]: Can you align reverse(logs) with the first arg above: "new TableRowAdapter...".

File: android/app/src/main/java/io/peggyjo/peggymeter/database/DataController.java [hunk 2]
```diff
@@ -81,4 +88,12 @@ public MoodAdapter getMoodAdapter() {
     public SettingAdapter getSettingAdapter() {
         return settingAdapter;
     }
+
+    public void clearMoodListeners() {
```
Comment (demetrNieA) [inline android/app/src/main/java/io/peggyjo/peggymeter/database/DataController.java]: Could you make it synchronized to make sure we won't have concurrent exception?

File: android/app/src/main/java/io/peggyjo/peggymeter/MainActivity.java [hunk 3]
```diff
@@ -58,9 +58,8 @@ private void showStatScreen() {
         }
         switch (currentMode) {
             case Graph:
-                if (historyGraphFragment == null) {
-                    historyGraphFragment = new HistoryGraphFragment();
-                }
+                dataController.clearMoodListeners();
+                historyGraphFragment = new HistoryGraphFragment();
```
Comment (rustema) [inline android/app/src/main/java/io/peggyjo/peggymeter/MainActivity.java]: Can you add a comment explaining why we re-create this fragment every time this function is called please?

File: android/app/src/main/java/io/peggyjo/peggymeter/MainActivity.java [hunk 4]
```diff
@@ -82,13 +81,11 @@ private void showStatScreen() {
                         .commit();
                 break;
             case Text:
-                if (historyTextFragment == null) {
-                    historyTextFragment = new HistoryTextFragment();
-                }
+                dataController.clearMoodListeners();
+                historyTextFragment = new HistoryTextFragment();
```
Comment (rustema) [inline android/app/src/main/java/io/peggyjo/peggymeter/MainActivity.java]: Ditto.

File: android/app/src/main/java/io/peggyjo/peggymeter/database/MoodAdapter.java [hunk 5]
```diff
@@ -24,14 +26,12 @@
...
-    private final DataController dataController;
     private final List<LogEntry> logs;
-    private final List<MoodListener> listeners;
+    private final Set<MoodListener> listeners;
 
     MoodAdapter(DataController dataController) {
-        this.dataController = dataController;
         this.logs = new ArrayList<>();
-        this.listeners = new ArrayList<>();
+        this.listeners = new HashSet<>();
```
Comment (demetrNieA) [inline android/app/src/main/java/io/peggyjo/peggymeter/database/MoodAdapter.java]: Do you have any specific need for a set here? If so please make sure we override hashCode and equals methods on implementation.

File: android/app/src/main/java/io/peggyjo/peggymeter/HistoryTextFragment.java (hunk 1 above)
Comment (vyakunin) [reply android/app/src/main/java/io/peggyjo/peggymeter/HistoryTextFragment.java]: done

File: android/app/src/main/java/io/peggyjo/peggymeter/MainActivity.java (hunk 3 above)
Comment (vyakunin) [reply android/app/src/main/java/io/peggyjo/peggymeter/MainActivity.java]: done

File: android/app/src/main/java/io/peggyjo/peggymeter/MainActivity.java (hunk 4 above)
Comment (vyakunin) [reply android/app/src/main/java/io/peggyjo/peggymeter/MainActivity.java]: done

File: android/app/src/main/java/io/peggyjo/peggymeter/database/MoodAdapter.java (hunk 5 above)
Comment (vyakunin) [reply android/app/src/main/java/io/peggyjo/peggymeter/database/MoodAdapter.java]: now I do

File: android/app/src/main/java/io/peggyjo/peggymeter/database/DataController.java (hunk 2 above)
Comment (vyakunin) [reply android/app/src/main/java/io/peggyjo/peggymeter/database/DataController.java]: done

File: android/app/src/main/java/io/peggyjo/peggymeter/database/MoodAdapter.java (hunk 5 above)
Comment (demetrNieA) [reply android/app/src/main/java/io/peggyjo/peggymeter/database/MoodAdapter.java]: Please do the same for the second adapter. Thanks!

**Pull request merged**